src/modules/did.js

```diff
@@ -0,0 +1,129 @@
+const ChluDID = require('chlu-did/src')
+const { getDigestFromMultihash } = require('../utils/ipfs')
+
+class ChluIPFSDID {
+
+    static isDIDID(didId) {
+        return typeof didId === 'string' && didId.indexOf('did:') === 0
+    }
+
+    constructor(chluIpfs) {
+        this.chluIpfs = chluIpfs 
+        // ChluDID instance
+        this.chluDID = new ChluDID()
+        // Well Known DIDs
+        // used for hardcoded DIDs in tests
+        // and will also be used for stuff like official Chlu trusted DIDs
+        this.wellKnownDIDs = {}
+        this.didId = null
+        this.publicDidDocument = null
+        this.privateKeyBase58 = null
+    }
+
+    async start() {
+        if (!this.isPresent()) {
+            // Generate a DID & Publish
+            await this.generate();
+            await this.publish(null, false)
+            await this.chluIpfs.persistence.persistData();
+        }
+    }
+
+    async generate() {
+        const did = await this.chluDID.generateDID()
+        return this.import(did)
+    }
+
+    import(did) {
+        this.publicDidDocument = did.publicDidDocument
+        this.didId = this.publicDidDocument.id
+        this.privateKeyBase58 = did.privateKeyBase58
+    }
+
+    export() {
+        return {
+            publicDidDocument: this.publicDidDocument,
+            privateKeyBase58: this.privateKeyBase58
+        }
+    }
+
+    isPresent() {
+        return this.publicDidDocument && this.didId && this.privateKeyBase58
+    }
+
+    async sign(data, privateKeyBase58) {
+        return this.chluDID.sign(privateKeyBase58 || this.privateKeyBase58, data)
+    }
+
+    async verify(didId, data, signature) {
+        const didDocument = await this.getDID(didId)
+        if (!didDocument) {
+            throw new Error(`Cannot verify signature by ${didId}: DID Document not found`)
+        }
+        return this.chluDID.verify(didDocument, data, signature)
+    }
+
+    async signMultihash(multihash, privateKeyBase58) {
+        const data = getDigestFromMultihash(multihash)
+        const result = await this.sign(data, privateKeyBase58)
+        return result.signature
+    }
+
+    async verifyMultihash(didId, multihash, signature) {
+        const data = getDigestFromMultihash(multihash)
+        return this.verify(didId, data, signature) 
+    }
+
+    async signPoPR(obj) {
```

Comment:
@fazo96 The did specific methods are more about creating did and maybe some used to sign any piece of data using the did lib.

All the other higher level application specific methods lke signPoPR, signReview etc can go in the crypto module and the crypto module can then use the did module to signMultihash or sign.*Data.

You see my thinking. It might help keep things clean.